### DISCUSSION TITLE
typescript support

### DISCUSSION THREAD

**Discussion by hasan-4** [tailwindlabs/tailwindcss#5311] [Ideas]
use ts instead of js in whole project

Example,
[windicss](https://windicss.org/)

**Comment by mikestopcontinues** [comment]
Yes! I'd love to be able to rely on typescript when working on `tailwind.config.ts`.

**Reply by lucasvazq** [reply]
A workaround is to transform the typescript files to commonjs, only if you can support it.

An example (yarn and swc):
```sh
yarn run swc ./tailwind.config.ts -o ./tailwind.config.js -C module.type=commonjs -C sourceMaps=inline
```

Recommendation: add this command to the build pipeline and add `tailwind.config.js` to the `.gitignore`.

**Comment by skimhugo** [comment]
Typescript config support is really a feature today
